src/ORM/AssociationCollection.php

```diff
@@ -114,9 +114,23 @@ public function keys()
      * @param string|array $class The type of associations you want.
      *   For example 'BelongsTo' or array like ['BelongsTo', 'HasOne']
      * @return array An array of Association objects.
+     * @deprecated 3.6.0 Use getByType() instead.
```

Comment:
Isn't that 3.5.3 then as well?

Reply:
The phpdoc [docs](https://docs.phpdoc.org/references/phpdoc/tags/deprecated.html) say:
> This tag MAY also contain a version number up till which it is guaranteed to be included in the software. Starting with the given version, the function will be removed or may be removed without further notice.

The method is gonna exist in 3.6 so wouldn't 3.6.0 be the right version?

Reply:
It will also exist in any 3.7 and alike. We cannot say anything about the future except for it will stay for this major release.
I think we always put the version number it got deprecated, since that is the most interesting information. We probably need to know at what point to look for in the migration guides and the rest of the code to use instead of it.